pkg/agent/docker_secrets_postrenderer.go

```diff
@@ -206,6 +204,25 @@ func getResourcePodSpec(resource map[interface{}]interface{}) map[interface{}]in
 			return nil
 		}
 		return podSpec
+	case "DaemonSet", "Deployment", "Job", "PodTemplate", "ReplicaSet", "ReplicationController", "StatefulSet":
+		// These resources all include a spec.template.spec PodSpec.
```

Comment:
what about a CronJob? 

Also, we should also cover the case in which the resource is a List of any of those components.

Reply:
> what about a CronJob?

Yes, `JobTemplateSpec` is in a CronJob also, I need to include this as a separate case.

Similarly a `PodTemplate` is in a `PodTemplateList` too, but...

> 
> Also, we should also cover the case in which the resource is a List of any of those resources.

...yes, this needs to be handled not here, but in the parent case iterating the resources. I'll do this in a follow-up PR.